doc/connectivity/networking/api/coap_client.rst

```diff
@@ -106,6 +106,52 @@ RFC7959 Figure 3: Block-Wise GET with Early Negotiation).
 
     ret = coap_client_req(&client, sock, &address, &req, -1);
 
+Optionally, the application can register a payload callback instead of providing a payload pointer
+for the CoAP upload. In such cases, the CoAP client library will call this callback when preparing
+a PUT/POST request, so that the application can provide the payload in blocks, instead of having to
+provide a single contiguous buffer with the entire payload. An example callback, providing the
+content of the Lorem Ipsum string can look like this:
+
+.. code-block:: c
```

Comment:
I hate to whine about this, but I dislike the fact that our code samples in the documentation do not adhere to our code style (tabs vs spaces). And it's an annoyance to do it properly, because rst file uses spaces for indentation, while our c code uses tabs.

I worked around this by using `literalinclude::` directives instead of inline code, but that isn't ideal either.

Reply:
Yeah but that probably would need to be brought to a broader audience, it's everywhere like this from what I see, it's not only this file.